java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -103,6 +103,8 @@ class EntityHandleManager(
         }
     }
 
+    override fun allStorageProxies() = singletonStorageProxies.values.plus(collectionStorageProxies.values)
+
```

Comment:
there's a slight risk here of these being modified while the copy is being made thus throwing a ConcurrentModificationException? Is Kotlin collection copy guaranteed lockless concurrency safe?

So you may need to make this `suspend fun allStorageProxies() = proxyMuxtex.withLock { ... }`